src/api/vbucket.h

```diff
@@ -18,20 +18,31 @@ namespace api {
 class VBucket {
  private:
   std::string name_;
+  VBucketID id_;
   std::list<std::pair<std::string, std::string>> linked_blobs_;
+  std::list<Trait *> attached_traits_;
+  Blob local_blob;
+  bool persist;
+  std::shared_ptr<Hermes> hermes_;
 
  public:
   /** internal Hermes object owned by vbucket */
-  std::shared_ptr<Hermes> hermes_;
-
-  VBucket(std::string initial_name, std::shared_ptr<Hermes> const &h)
-    : name_(initial_name), hermes_(h) {
+  VBucket(std::string initial_name, std::shared_ptr<Hermes> const &h,
+          bool persist, Context ctx)
+      : name_(initial_name),
+        id_({0, 0}),
+        linked_blobs_(),
+        attached_traits_(),
+        local_blob(),
+        persist(persist),
+        hermes_(h) {
     LOG(INFO) << "Create VBucket " << initial_name << std::endl;
-    if (hermes_->vbucket_list_.find(initial_name) ==
-        hermes_->vbucket_list_.end())
-      hermes_->vbucket_list_.insert(initial_name);
-    else
-      std::cerr << "VBucket " << initial_name << " exists\n";
+    (void)ctx;
+    if (IsBucketNameTooLong(name_)) {
```

Comment:
Shouldn't this be `IsVBucketNameTooLong`?.

Reply:
FIXED